### DISCUSSION TITLE
Use the Filter API to query a reference

### DISCUSSION THREAD

**Discussion by heruan** [MorphiaOrg/morphia#1978] [General]
I've built many repositories using Morphia's Filter API and now I need to query a reference field like this:

```js
db.Foo.aggregate(
  {$unwind: "$bars"},
  {$lookup: {
    from:"bar",
    localField: "bars",
    foreignField: "_id",
    as: "bar"
   }},
   {$match: {
    "bar.testprop": true
   }}
)
``` 

(Context here: https://stackoverflow.com/a/39476690/1006669)

How can I express a query like this resulting in a `Filter` object?

**Comment by evanchooly** [comment]
What you're looking for there is the [aggregations API](https://morphia.dev/morphia/2.2/aggregations.html).  Your resultant Java code should look a lot like what you have above modulo a few Java syntax variations.  As for querying against that reference, all you should need to do is pass in a (java) reference to the object you want to query against.  If you don't have that reference handy (which is likely), what you pass in can vary based on how you've mapped your entities.  If you have that (morphia) reference configured to be `idOnly`, you need only pass in the ID value for that remote document.  If you don't have that option enabled, you'll need to pass a [DBRef](https://mongodb.github.io/mongo-java-driver/4.7/apidocs/mongodb-driver-core/com/mongodb/DBRef.html#%3Cinit%3E(java.lang.String,java.lang.Object)) instance in your `Filter`.
